lib/ttnt/testtask.rb

```diff
@@ -0,0 +1,35 @@
+module TTNT
+  class TestTask
+    @@instances = []
+
+    def self.instances
+      @@instances
+    end
+
+    ATTRIBUTES = %i(
+      name
+      libs
+      pattern
+      options
+      test_files
+      warning
+      loader
+      ruby_opts
+      description
+      pattern
+    ).freeze
+
+    ATTRIBUTES.each do |attr|
+      attr_reader attr
+    end
+
+    def initialize(rake_test_task)
+      ATTRIBUTES.each do |attr|
```

Comment:
Actually the `respond_to?` check is better, you're right ; with the `- [:@test_files]` we will have the same problem that we had with the `ATTRIBUTES` constant! :+1: